### ISSUE TITLE
:DB hangs in GVim when attempting to open interactive console

### ISSUE THREAD

**Issue by dexterous** [tpope/vim-dadbod#51]
Per README, if only connection URL is provided and SQL string / file is omitted, it should spawn an interactive console. However, when executed in GVim, the Vim just locks up.

I have tpope/vim-dispatch installed and running...
```
Start postgresql://...
```
... does open `psql` in interactive mode (albeit) within the Ex bar.

With a little experimentation, I found that taking off the `silent` at https://github.com/tpope/vim-dadbod/blob/master/autoload/db.vim#L206
```
silent execute 'Start' escape(cmd, '%#')
```
... does the trick.

**Comment by tpope** [comment]
That's unexpected. Does `:silent Start cmd.exe` lock up the same way?

**Comment by vadz** [comment]
FWIW this bug is still present when using the current version (i.e. fe5a55e (Add Snowflake support (#174), 2024-10-18)) with gvim 9.0.1677 under Unix.

Removing `silent` fixes it.

**Comment by tpope** [comment]
If it's the `Start` line causing it, it's probably a dispatch.vim issue. When you delete the `silent` to make it work, what adapter does it use?

**Comment by vadz** [comment]
Sorry for being clueless, but how do I find out which adapter is used, exactly?

**Comment by tpope** [comment]
What happens when you run it?

**Comment by vadz** [comment]
If I run it without changing the plugin, GVim just hangs, with `:DB sqlite3://...` in the command line and nothing else happening until I (blindly) type `.q<Enter>`, after which it gets back to normal mode (`Ctrl-C` doesn't work).

If I remove `silent`, it shows SQLite prompt which I can use more or less normally (except the usual readline keys don't work).

FWIW I think I'm not using any adapter at all, i.e. `exists(':Start:')` returns 0 for me.

**Comment by tpope** [comment]
> FWIW I think I'm not using any adapter at all, i.e. `exists(':Start:')` returns 0 for me.

Is that a typo? The correct check is `exists(':Start')`.

**Comment by tpope** [comment]
Okay, yeah, distinct issue but possibly with the same cause. I bet this is contingent on whether or not `'guioptions'` contains `!`.

**Comment by vadz** [comment]
You're right, my `guioptions` is set to `r` and adding `!` to it lets `:DB` work even with `silent`, thanks!
